plugins/backend/mock/mock-plugin/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"os"
+
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/logical/plugin"
+	"github.com/hashicorp/vault/plugins/backend/mock"
+)
+
+func main() {
+	apiClientMeta := &pluginutil.APIClientMeta{}
+	flags := apiClientMeta.FlagSet()
+	flags.Parse(os.Args)
+	tlsConfig := apiClientMeta.GetTLSConfig()
+	tlsProviderFunc := pluginutil.VaultPluginTLSProvider(tlsConfig)
+
+	plugin.Serve(&plugin.ServeOpts{
```

Comment:
I went with `plugins.Serve()` since I thought I would reduce the number of imports needed for the compiled plugin binary (non-db plugins wouldn't import `dbplugin`, for instance).